bash/drive-utils.bash

```diff
@@ -0,0 +1,529 @@
+#!/usr/bin/env bash
+
+###################################################
+# Used in collecting file properties from output json after a file has been uploaded/cloned
+# Also handles logging in log file if LOG_FILE_ID is set
+# Globals: 1 variables, 2 functions
+#   Variables - LOG_FILE_ID
+#   Functions - _error_logging, _json_value
+# Arguments: 1
+#   ${1} = output jsom
+# Result: set fileid and link, save info to log file if required
+###################################################
+_collect_file_info() {
+    declare json="${1}" info
+    FILE_ID="$(_json_value id 1 1 <<< "${json}")"
+    [[ -z ${FILE_ID} ]] && _error_logging "${2}" && return 1
+    [[ -z ${LOG_FILE_ID} && ! -d ${LOG_FILE_ID} ]] && return 0
+    info="$(
+        printf "%s\n" "Link: https://drive.google.com/open?id=${FILE_ID}"
+        printf "%s\n" "Name: $(_json_value name 1 1 <<< "${json}")"
+        printf "%s\n" "ID: ${FILE_ID}"
+        printf "%s\n\n" "Type: $(_json_value mimeType 1 1 <<< "${json}")"
+    )"
+    printf "%s\n" "${info}" >> "${LOG_FILE_ID}"
+}
+
+###################################################
+# Error logging wrapper
+###################################################
+_error_logging() {
+    "${QUIET:-_print_center}" "justify" "Upload ERROR" ", ${1:-} not ${STRING:-uploaded}." "=" 1>&2
+    printf "\n\n\n" 1>&2
+}
+
+###################################################
+# Get information for a gdrive folder/file.
+# Globals: 2 variables, 1 function
+#   Variables - API_URL, API_VERSION
+#   Functions - _json_value
+# Arguments: 3
+#   ${1} = folder/file gdrive id
+#   ${2} = information to fetch, e.g name, id
+#   ${3} = Access Token
+# Result: On
+#   Success - print fetched value
+#   Error   - print "message" field from the json
+# Reference:
+#   https://developers.google.com/drive/api/v3/search-files
+###################################################
+_drive_info() {
+    [[ $# -lt 3 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare folder_id="${1}" fetch="${2}" token="${3}"
+    declare search_response
+
+    "${EXTRA_LOG:-:}" "justify" "Fetching info.." "-" 1>&2
+    search_response="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)